ThunkLibs/libOpenCL/libOpenCL_Host.cpp

```diff
@@ -0,0 +1,595 @@
+/*
+$info$
+tags: thunklibs|X11
+desc: Handles callbacks and varargs
+$end_info$
+*/
+
+#include <cstdlib>
+#include <stdio.h>
+
+#include <dlfcn.h>
+#include <unordered_map>
+#include <shared_mutex>
+
+#include "common/CrossArchEvent.h"
+#include "common/Host.h"
+#include <vector>
+
+#include "libOpenCL_Common.h"
+
+#include "thunkgen_host_libOpenCL.inl"
+
+// clCreateContext - pfn_notify - Is live until context is destroyed (with refcount)
```

Comment:
This is missing a heading comment explaining why the thunk library needs to worry about lifetimes in the first place.

Reply:
Added a comment block about this.